analyzer/codechecker_analyzer/analyzers/result_handler_base.py

```diff
@@ -90,10 +90,18 @@ def analyzer_result_file(self):
         if not self.__result_file:
             analyzed_file_name = os.path.basename(self.analyzed_source_file)
 
-            build_info = str(self.buildaction.analyzer_type) + '_' + \
+            source_file = os.path.normpath(
+                os.path.join(self.buildaction.directory,
+                             self.analyzed_source_file))
+
+            source_file = os.path.relpath(source_file,
```

Comment:
Wouldn't it be possible to simply use the absolute path of the source file for the hash? I mean, if the collision is because of the same basename of files, and we add the user-home to that, it would be still not unique. And if the uniqueness is ensured later, this expansion is redundant IMHO.